media/multipart/common/src/test/java/io/helidon/media/multipart/common/MultiPartDecoderTest.java

```diff
@@ -0,0 +1,571 @@
+        decoder.subscribe(testSubscriber);
+        assertThat(testSubscriber.complete, is(equalTo(false)));
+        assertThat(testSubscriber.error, is(notNullValue()));
+        assertThat(testSubscriber.error.getMessage(), is(equalTo("oops")));
+    }
+
+    @Test
+    public void testSubcribingMoreThanOnce() {
+        MultiPartDecoder decoder = MultiPartDecoder.create("boundary", MEDIA_CONTEXT.readerContext());
+        chunksPublisher("foo".getBytes()).subscribe(decoder);
+        try {
+            chunksPublisher("bar".getBytes()).subscribe(decoder);
+            fail("exception should be thrown");
+        } catch(IllegalStateException ex) {
+            assertThat(ex.getMessage(), is(equalTo("Input subscription already set")));
+        }
+    }
+
+    /**
+     * Types of test subscribers.
+     */
+    enum SUBSCRIBER_TYPE {
+        INFINITE,
+        ONE_BY_ONE,
+        CANCEL_AFTER_ONE,
+    }
+
+    /**
+     * A part test subscriber.
+     */
+    static class BodyPartSubscriber implements Subscriber<BodyPart>{
+
+        private final SUBSCRIBER_TYPE subscriberType;
+        private final Consumer<BodyPart> consumer;
+        private Subscription subcription;
+        private Throwable error;
+        private boolean complete;
+
+        BodyPartSubscriber(SUBSCRIBER_TYPE subscriberType, Consumer<BodyPart> consumer) {
+            this.subscriberType = subscriberType;
+            this.consumer = consumer;
+        }
+
+        @Override
+        public void onSubscribe(Subscription subscription) {
+            this.subcription = subscription;
+            if (subscriberType == SUBSCRIBER_TYPE.INFINITE) {
+                subscription.request(Long.MAX_VALUE);
+            } else {
+                subscription.request(1);
+            }
+        }
+
+        @Override
+        public void onNext(BodyPart item) {
+            if (consumer == null){
+                return;
+            }
+            consumer.accept(item);
+            if (subscriberType == SUBSCRIBER_TYPE.ONE_BY_ONE) {
+                subcription.request(1);
+            } else if (subscriberType == SUBSCRIBER_TYPE.CANCEL_AFTER_ONE) {
+                subcription.cancel();
+            }
+        }
+
+        @Override
+        public void onError(Throwable ex) {
+            error = ex;
+        }
+
+        @Override
+        public void onComplete() {
+            complete = true;
+        }
+    }
+
+    /**
+     * Create the parts publisher for the specified boundary and request
+     * chunks.
+     * @param boundary multipart boundary string
+     * @param chunks request chunks
+     * @return publisher of body parts
+     */
+    static Publisher<? extends BodyPart> partsPublisher(String boundary, byte[]... chunks) {
+        MultiPartDecoder decoder = MultiPartDecoder.create(boundary, MEDIA_CONTEXT.readerContext());
+        chunksPublisher(chunks).subscribe(decoder);
```

Comment:
We definitely need `Multi.just(...).via(Flow.Processor).subscribe(Flow.Subscriber)` like MPRS have

Reply:
Yes, this would simplify using processors.